benchmark-service/benchmark/benchmark.py

```diff
@@ -272,6 +247,93 @@ async def update_commits(app):
     }
 
 
+async def update_commit(app, sha):  # pylint: disable=unused-argument
+    github_client = app['github_client']
+    batch_client = app['batch_client']
+    gs_reader = app['gs_reader']
+
+    file_path = f'{BENCHMARK_RESULTS_PATH}/{sha}.json'
+
+    request_string = f'/repos/hail-is/hail/commits/{sha}'
+
+    gh_commit = await github_client.getitem(request_string)
+
+    has_results_file = gs_reader.file_exists(file_path)
+    batch_statuses = [b._last_known_status async for b in batch_client.list_batches(q=f'sha={sha}')]
+    complete_batch_statuses = [bs for bs in batch_statuses if bs['complete']]
+    running_batch_statuses = [bs for bs in batch_statuses if not bs['complete']]
+
+    if has_results_file:
+        assert complete_batch_statuses, batch_statuses
+        log.info(f'commit {sha} has a results file')
+        status = complete_batch_statuses[-1]
+    elif running_batch_statuses:
+        status = running_batch_statuses[-1]
+        log.info(f'batch already exists for commit {sha}')
+    else:
+        batch_id = await submit_test_batch(batch_client, sha)
+        batch = await batch_client.get_batch(batch_id)
+        status = batch._last_known_status
+        log.info(f'submitted a batch {batch_id} for commit {sha}')
+
+    commit = {
+        'sha': sha,
+        'title': gh_commit['commit']['message'],
+        'author': gh_commit['commit']['author']['name'],
+        'date': gh_commit['commit']['author']['date'],
+        'status': status
+    }
+    return commit
+
+
+@router.get('/api/v1alpha/benchmark/commit/{sha}')
+async def get_status(request):  # pylint: disable=unused-argument
+    global benchmark_data
+    sha = str(request.match_info['sha'])
+
+    app = request.app
+    batch_client = app['batch_client']
+    batch_statuses = [b._last_known_status async for b in batch_client.list_batches(q=f'sha={sha}')]
+
+    try:
+        status = batch_statuses[-1]
+        log.info('got commit status')
```

Comment:
I was thinking that this should return the same thing as the full commit and not just the status. I'm also not sure you need to wrap the status in `{'status': status}` and could just return the commit directly.

```
commit = {
         'sha': sha,
         'title': gh_commit['commit']['message'],
         'author': gh_commit['commit']['author']['name'],
         'date': gh_commit['commit']['author']['date'],
         'status': status
     }
```

To do this, you might need to split `update_commit` into two functions one of which is `get_commit`. This should look like this:

```python3
async def get_commit(app, sha):  # pylint: disable=unused-argument
     github_client = app['github_client']
     batch_client = app['batch_client']
     gs_reader = app['gs_reader']

      file_path = f'{BENCHMARK_RESULTS_PATH}/{sha}.json'

      request_string = f'/repos/hail-is/hail/commits/{sha}'

      gh_commit = await github_client.getitem(request_string)

    has_results_file = gs_reader.file_exists(file_path)
     batch_statuses = [b._last_known_status async for b in batch_client.list_batches(q=f'sha={sha}')]
     complete_batch_statuses = [bs for bs in batch_statuses if bs['complete']]
     running_batch_statuses = [bs for bs in batch_statuses if not bs['complete']]

      if has_results_file:
         assert complete_batch_statuses, batch_statuses
         log.info(f'commit {sha} has a results file')
         status = complete_batch_statuses[-1]
     elif running_batch_statuses:
         status = running_batch_statuses[-1]
         log.info(f'batch already exists for commit {sha}')
     else:
         status = None
         log.info(f'no batches or results file exists for {sha}')

      commit = {
         'sha': sha,
         'title': gh_commit['commit']['message'],
         'author': gh_commit['commit']['author']['name'],
         'date': gh_commit['commit']['author']['date'],
         'status': status
     }
     return commit
```

```python3
async def update_commit(app, sha):  # pylint: disable=unused-argument
     commit = await get_commit(app, sha)
     if commit['status'] is not None:
           return commit

     batch_client = app['batch_client']

     batch_id = await submit_test_batch(batch_client, sha)
     batch = await batch_client.get_batch(batch_id)
     commit['status'] = batch._last_known_status
     log.info(f'submitted a batch {batch_id} for commit {sha}')

     return commit
```